src/test.rs

```diff
@@ -192,15 +199,95 @@ fn it_works_decode_with_version_0() {
     CENNZnutV0::decode(&mut &encoded[..]).unwrap();
 }
 
+#[test]
+fn it_works_encode_with_constraints() {
+    let contract = Contract {
+        data_table: DataTable::new(vec![
+            PactType::Numeric(Numeric(111)),
+            PactType::Numeric(Numeric(333)),
+            PactType::StringLike(StringLike(b"testing")),
+        ]),
+        bytecode: [OpCode::EQ.into(), 0, 0, 1, 0, OpCode::EQ.into(), 0, 1, 1, 1].to_vec(),
+    };
+    let mut constraints: Vec<u8> = Vec::new();
+    contract.encode(&mut constraints);
+
+    let method = Method {
+        name: "method_test".to_string(),
+        block_cooldown: None,
+        constraints: Some(constraints),
+    };
+    let mut methods: Vec<(String, Method)> = Default::default();
+    methods.push((method.name.clone(), method.clone()));
+
+    let module = Module {
+        name: "module_test".to_string(),
+        block_cooldown: None,
+        methods,
+    };
+
+    let mut modules: Vec<(String, Module)> = Default::default();
+    modules.push((module.name.clone(), module.clone()));
+
+    let cennznut = CENNZnutV0 { modules };
+
+    assert_eq!(
+        cennznut.encode(),
+        vec![
+            0, 0, 0, 64, 109, 111, 100, 117, 108, 101, 95, 116, 101, 115, 116, 0, 0, 0, 0, 0, 0, 0,
+            0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 74, 109, 101, 116, 104, 111, 100, 95, 116,
+            101, 115, 116, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 192,
+            128, 16, 246, 0, 0, 0, 0, 0, 0, 0, 128, 16, 178, 128, 0, 0, 0, 0, 0, 0, 0, 224, 116,
+            101, 115, 116, 105, 110, 103, 5, 0, 0, 1, 0, 5, 0, 1, 1, 1,
+        ]
+    );
+}
+
+#[test]
+fn it_works_decode_with_constraints() {
+    let encoded_cennznut: Vec<u8> = vec![
+        0, 0, 0, 64, 109, 111, 100, 117, 108, 101, 95, 116, 101, 115, 116, 0, 0, 0, 0, 0, 0, 0, 0,
+        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 74, 109, 101, 116, 104, 111, 100, 95, 116, 101, 115,
+        116, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 192, 128, 16, 246,
+        0, 0, 0, 0, 0, 0, 0, 128, 16, 178, 128, 0, 0, 0, 0, 0, 0, 0, 224, 116, 101, 115, 116, 105,
+        110, 103, 5, 0, 0, 1, 0, 5, 0, 1, 1, 1,
+    ];
+    CENNZnutV0::decode(&mut &encoded_cennznut[..]).unwrap();
```

Comment:
This test should actually check that `encoded_cennznut` when decoded is equal to the expected `CENNZnutV0` struct as decoding the bytes doesn't necessarily ensure correctness